### ISSUE TITLE
merge_records writes header records to memory

### ISSUE THREAD

**Issue by Suudy** [TexZK/hexrec#24]
In merge_records, there is:

```Python
    blocks = []
    for records in data_records:
        blocks.extend([r.address, r.data] for r in records)
    blocks = Memory.collapse_blocks(blocks)

```

But this doesn't filter out the header records, which leads to header data corrupting memory.

**Comment by TexZK** [comment]
Hi Suudy, thanks for the feedback, I'm going to integrate your PR soon.

FYI: That function was made to be provided *data* records only, hence the *data_* prefixes and the docstring.
Anyway, I agree that the filtering can be done easily within this function itself.

**Comment by TexZK** [comment]
Merged into commit a1d8bc0d59e4ac784cf0697867641656536d04d6
